lib/modulesync/git.rb

```diff
@@ -19,16 +19,18 @@ def self.pull(git_base, name, branch, opts)
 
       # Repo already cloned, check out master and override local changes
       else
-        puts "Overriding any local changes to repositories in #{PROJ_ROOT}"
-        repo = ::Git.open("#{PROJ_ROOT}/#{name}")
-        repo.branch(branch).checkout
-        repo.reset_hard
-        repo.pull('origin', branch)
+        if ! git_provider.eql? 'gerrit'
+          puts "Overriding any local changes to repositories in #{PROJ_ROOT}"
+          repo = ::Git.open("#{PROJ_ROOT}/#{name}")
+          repo.branch(branch).checkout
+          repo.reset_hard
+          repo.pull('origin', branch)
+        end
       end
     end
 
     # Git add/rm, git commit, git push
-    def self.update(name, files, message, branch)
+    def self.update(name, files, message, branch, git_provider, git_user, git_provider_address, git_provider_port)
```

Comment:
This won't work if the user's supplied the full git_base instead of the individual git_* parameters.